extensions/NexusKitten/sgrab.js

```diff
@@ -0,0 +1,211 @@
+                }
+              }
+            },
+            '---',
+            {
+              opcode: 'projectgrab',
+              blockType: Scratch.BlockType.REPORTER,
+              text: 'grab [WHAT] count of project id [WHO]',
+              arguments: {
+                WHAT: {
+                  type: Scratch.ArgumentType.STRING,
+                  menu: 'WHAT3'
+                },
+                WHO: {
+                  type: Scratch.ArgumentType.STRING,
+                  defaultValue: '717954208'
+                }
+              }
+            },
+            {
+                opcode:'rankprojectgrab',
+                blockType: Scratch.BlockType.REPORTER,
+                text: 'global [WHAT] ranking for project id [WHO]',
+                arguments: {
+                  WHAT: {
+                    type: Scratch.ArgumentType.STRING,
+                    menu: 'WHAT4'
+                  },
+                  WHO: {
+                    type: Scratch.ArgumentType.STRING,
+                    defaultValue: '717954208'
+                  }
+                }
+            },
+            {
+                opcode:'idtoname',
+                blockType: Scratch.BlockType.REPORTER,
+                text: 'name of project id [WHO]',
+                arguments: {
+                  WHO: {
+                    type: Scratch.ArgumentType.STRING,
+                    defaultValue: '717954208'
+                  }
+                }
+            },
+            {
+                opcode:'idtoowner',
+                blockType: Scratch.BlockType.REPORTER,
+                text: 'creator of project id [WHO]',
+                arguments: {
+                  WHO: {
+                    type: Scratch.ArgumentType.STRING,
+                    defaultValue: '717954208'
+                  }
+                }
+            },
+          ],
+          menus: {
+            WHAT: {
+              acceptReporters: true,
+              items: ['follower', 'following']
+            },
+            WHAT2: {
+              acceptReporters: true,
+              items: ['follower', 'love', 'favorite', 'view']
+            },
+            WHAT3: {
+                acceptReporters: true,
+                items: ['love', 'favorite', 'view']
+            },
+            WHAT4: {
+              acceptReporters: true,
+              items: ['love', 'favorite', 'view']
+            },
+            WHAT5: {
+              acceptReporters: true,
+              items: ['about me', 'wiwo', 'location', 'status']
+            }
+          }
+        };
+      }
+      async usergrab(args) {
+        const response = await Scratch.fetch('https://scratchdb.lefty.one/v3/user/info/'+args.WHO);
+        const jsonData = await response.json();
+        if (args.WHAT === 'follower') {
+          return jsonData.statistics.followers;
+        } else if (args.WHAT === 'following') {
+          return jsonData.statistics.following;
+        }

```

Comment:
if args.WHAT is something you don't recognize, all of the blocks still needs to return *something*, probably just an empty string